### DISCUSSION TITLE
recent-updates-for-netescapades-enumgenerators-interceptors/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#441] [Announcements]
# recent-updates-for-netescapades-enumgenerators-interceptors/

In this post I describe the experimental interceptor support in NetEscapades.EnumGenerators that replaces ToString() calls with ToStringFast() automatically

https://andrewlock.net/recent-updates-for-netescapades-enumgenerators-interceptors/

**Comment by a14907** [comment]
I have a question! How do you know the location of the method:ToString(). Because the Interceptor need you tell the method location in the source generat stage , but I have not find the way to get the location! How to do that!

**Comment by Reonekot** [comment]
About the [Description]/[Display] thing, as a user of the library I agree. Even with ToStringFast() it is unexpected (if you do not read the docs) that it doesn't return the same output as ToString() - ofc. nice with an easy opt. in on ToStringFast() and/or the attribute. (Actually not sure which I'd prefer - I'd probably go for being explicit at the callsite vs. "global" opt-in on it.)

A shame string interpolation version cant be caught. I'm pretty sure I'm going to refactor my own code 3 months after it was written from $"{Color.Red.ToString()}" -> $"{Color.Red}" just because I would have forgotten why .ToString() was there. :S For that fact, I'd probably consider keep using ToStringFast() in those cases at least.
